### DISCUSSION TITLE
ListingsItems patch_listings_item is accepted but SKU not updated

### DISCUSSION THREAD

**Discussion by MyGuyCai** [saleweaver/python-amazon-sp-api#656] [Q&A] [answered]
I am trying to create a function that changes the active price of a SKU, the response says the body is accepted but my live listings are not changed.

Here is the function I am using:
```
def change_one_product(seller_sku, price):
    dictionary = {
        "productType": "PRODUCT",
        "patches": [
            {
                "op": "replace",
                "path": "/attributes/purchasable_offer",
                "value": [
                    {
                        "marketplace_id": marketplace_uk[0],
                        "currency": "GBP",
                        "our_price": [{"schedule": [{"value_with_tax": price}]}]
                    }
                ]
            }
        ]
    }

    res = ListingsItems(credentials=credentials, marketplace=Marketplaces.UK).patch_listings_item(
        sku=seller_sku,
        sellerId=seller_id,
        marketplaceIds=marketplace_uk,
        body=dictionary
    )

    return res
```

The returned payload is:
```
'payload': {'issues': [],
             'sku': 'ABC16510',
             'status': 'ACCEPTED',
             'submissionId': 'XXXX'}
```
Is there a way to check the status using the submissionId?  
Also is there somewhere where I can find more detailed documentation on how to write the body?

**Comment by denisneuf** [comment]
I checked and I have some function similar like you. The payload message you get looks good too, just be aware of:
- Change the price is not automatically refreshed in the seller central it could take a while to show in the inventory page.
- Price changed in the listing can take even longer time to be reflected, 15 minutes or later.
What I do is to record any patch I submitted with no issue and is 'ACCEPTED' and check after 30 minutes or more.

**Reply by MyGuyCai** [reply]
You seem correct, checking again the price has updated about 40 mins later.
Do you know where I can find documentation on how to write the body, say for changing country of origin, tax code etc etc?

**Comment by denisneuf** [comment, marked as answer]
About your question: _Is there a way to check the status using the submissionId?_
I think I remember I read they will develop in the future some system to do that, think it will be like the reports in the Amazon Advertising API, you submit the submissionId in some endpoint and will get the payload "IN_PROGRESS", "DONE",...ETC but with Amazon api's you never know when or if finally will implement as they are very "particular" how they treat the "listings".

_Do you know where I can find documentation on how to write the body, say for changing country of origin, tax code etc etc?_
Well there is no official documentation or I couldn't find in the [official documentation](https://developer-docs.amazon.com/sp-api/)

What I did is use the ProductTypeDefinitions endpoint (from sp_api.api import ProductTypeDefinitions) if you use this endpoint it will lead you to a json with the property names by propertyGroups, I will post just a small portion as there quite long:

'product_details': 
{'description': 'Detalles del producto', 
'propertyNames': [
'product_description',
'bullet_point',
'generic_keyword',
'style',
'material',
...
]

And in the bottom of this endless json, there is a link to the json model, something like this:

             'schema': {'checksum': 'eVdAaCaMJHdRHSIAI53/Wg==',
              'link': {'resource': 'https://selling-partner-definitions-prod-dub.s3.eu-west-1.amazonaws.com/schema/ELECTRONIC_COMPONENT.json/ugly-amz-security-token',
                                 'verb': 'GET'}}},

So if you download that json and use some kind of viewer to make it human readable you can find more or less the definition of the properties.

![json](https://user-images.githubusercontent.com/16271058/181399113-7d2414a0-251e-424a-99b0-133a5337a12a.png)

Some tips I found using the patching:

- The main attributes you can use a same function, as the only thing you need change is the path and the value:

    option = "add" # add, replace
    attribute = "color"  # country_of_origin, product_description, generic_keyword
    path = "/attributes/%s" % attribute
    value = "AW 335"

then I build the dictionary:

    dictionary = {}
    dictionary["productType"] = "PRODUCT"
    op = {
        "op": option,
        "path": path
    }
    dictionary["patches"] = [op]
    options = []
    op = {"value": value, "marketplace_id": marketplace_id}
    options.append(op)
    dictionary["patches"][0]["value"] = options
    logging.info(dictionary)

- Other specific attributes like the one you start the question, price, bullet_points,...need a specific way to submit the values.

- Other obvious attributes like "size" are not available from the API, you need use the edit inventory page in the seller central.

- Some supported attributes like "brand" you can submit and get "ACCEPTED" but will never change because the way Amazon deal with the listings prevent to change a brand of a listing that has been created.

As the way Amazon treat the listings, there is not an owner of a listing and different sellers can submit different updates, the most easy example is the Title, should be easy to change the title as is the kind of regular attribute you may need change but you will face this problem, the submission is accepted, in your inventory page it shows the new title, but the listing will not be reflected because Amazon is blocking that attribute to being modified.

And not all, just some listings, so doing the same the result can be success or fail : (

So I am looking forward for that system that says, submitted successfully and "ACCEPTED" or "REJECTED" but...you know ... Amazon ...

Kindest regards,
D

**Reply by MyGuyCai** [reply]
Thank you ProductTypeDefinitions is exactly what I was looking for!
I had already looked through the docs but was unable to find anything about the json values
